internal/controller/postgrescluster/util.go

```diff
@@ -285,3 +287,61 @@ func safeHash32(content func(w io.Writer) error) (string, error) {
 	}
 	return rand.SafeEncodeString(fmt.Sprint(hash.Sum32())), nil
 }
+
+// AdditionalVolumeMount returns the name and mount path of the additional volume.
+func AdditionalVolumeMount(volumeName, claimName string) corev1.VolumeMount {
+	return corev1.VolumeMount{
+		Name:      claimName,
+		MountPath: "/volumes" + "/" + volumeName}
+}
+
+// addAdditionalVolumesToSpecifiedContainers adds additional volumes to the specified
+// containers in the specified pod
+func addAdditionalVolumesToSpecifiedContainers(template *corev1.PodTemplateSpec,
+	additionalVolumes []*v1beta1.AdditionalVolume) {
+
+	for _, additionalVolumeRequest := range additionalVolumes {
+
+		additionalVolumeMount := AdditionalVolumeMount(
+			additionalVolumeRequest.Name,
+			additionalVolumeRequest.ClaimName)
+
+		additionalVolume := corev1.Volume{
+			Name: additionalVolumeMount.Name,
```

Comment:
🤔 Ah, hmm. Volume names must be unique in a pod. What happens If someone uses `postgres-data` for their volume name? Maybe we should prefix with `volume-`?

🤔 🤔 What other (Pod) validation happens on this name in K8s? Maybe a limit on the length?

https://github.com/CrunchyData/postgres-operator/blob/6a30d727f64ab088c32cc520bbae0ce411fb4e56/internal/postgres/reconcile.go#L26-L28

Reply:
For the pod.spec.volumes, its says "Must be a DNS_LABEL and unique within the pod." ([cite](https://github.com/kubernetes/api/blob/v0.33.3/core/v1/types.go#L38))

That's what the code comment says, at least.

+1 for a prefix that prevents other problems

Reply:
... and if we're prepending something to the volume name, then the max length on our volume name field is less than upstream.